src/routers/institutions.py

```diff
@@ -38,6 +38,7 @@ async def set_db(request: Request, session: Annotated[AsyncSession, Depends(get_
 
 @router.get("/", response_model=List[FinancialInstitutionWithRelationsDto])
 @requires("authenticated")
+@fi_search_association_check
```

Comment:
went the decorator / annotation route, how do we feel about this vs dependencies?

Reply:
Looks pretty slick to me